crawl-ref/source/item-use.cc

```diff
@@ -2872,14 +2875,21 @@ void read(item_def* scroll)
 
     // if we have blurry vision, we need to start a delay before the actual
     // scroll effect kicks in.
-    if (you.get_mutation_level(MUT_BLURRY_VISION))
+    // quill orc fast reading offsets one level of blurry vision.
+    if (you.get_mutation_level(MUT_BLURRY_VISION) - qo > 0)
     {
         // takes 0.5, 1, 2 extra turns
-        const int turns = max(1, you.get_mutation_level(MUT_BLURRY_VISION) - 1);
+        const int turns = max(1, you.get_mutation_level(MUT_BLURRY_VISION) - qo - 1);
         start_delay<BlurryScrollDelay>(turns, *scroll);
-        if (you.get_mutation_level(MUT_BLURRY_VISION) == 1)
+        if (you.get_mutation_level(MUT_BLURRY_VISION) - qo == 1)
             you.time_taken /= 2;
     }
+    // quill orcs without blurry vision read scrolls faster
+    else if (you.species == SP_QUILL_ORC && !you.get_mutation_level(MUT_BLURRY_VISION))
+    {
+        you.time_taken /= 2;
```

Comment:
does this result in scrolls reading in 0 auts if you read them without blurry vision? Should this use `div_rand_round` instead?

Reply:
I bug-tested this before making the PR, and it results in reading in 0.5 auts, as intended (I think you.time_taken is AUTSx10). I also made sure that it interacts correctly with blurry vision and spell memorization.